src/CacheInvalidator.php

```diff
@@ -47,28 +48,38 @@ class CacheInvalidator
     const INVALIDATE = 'invalidate';
 
     /**
+     * Value to check support of tag invalidation operation.
+     */
+    const TAG = 'tag';
+
+    /**
      * @var ProxyClientInterface
      */
     private $cache;
 
     /**
-     * @var EventDispatcherInterface
+     * @var TagHandler
      */
-    private $eventDispatcher;
+    private $tagHandler;
 
     /**
-     * @var string
+     * @var EventDispatcherInterface
      */
-    private $tagsHeader = 'X-Cache-Tags';
+    private $eventDispatcher;
 
     /**
      * Constructor
      *
-     * @param ProxyClientInterface $cache HTTP cache
+     * @param ProxyClientInterface $cache      HTTP cache
+     * @param TagHandler           $tagHandler A tag handler instance (optional)
      */
-    public function __construct(ProxyClientInterface $cache)
+    public function __construct(ProxyClientInterface $cache, TagHandler $tagHandler = null)
```

Comment:
But then by what criteria do we say that the ObjectHandler shouldn't be in this facade? or should it? (I would lean towards not)